From 0ff3599d918cd944b3cd4fe3e5b4906b42a18273 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Tue, 7 May 2013 22:11:42 +0200
Subject: [PATCH 1/2] modalDialog: Replace Shell.Stack with layout manager

Commit e98eb57e3e8b added flags to expand the dialog's background
stack, which works fine with the current clutter-1.16 branch, but
breaks on clutter-1.14 (as shipped with GNOME 3.8).
Using an St.Widget with a Clutter.BinLayout fixes this, and is more
modern Clutter usage.

https://bugzilla.gnome.org/show_bug.cgi?id=699877
---
 js/ui/modalDialog.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/ui/modalDialog.js b/js/ui/modalDialog.js
index 84c5c3b..c85f9a3 100644
--- a/js/ui/modalDialog.js
+++ b/js/ui/modalDialog.js
@@ -65,7 +65,7 @@ const ModalDialog = new Lang.Class({
         this._group.connect('key-press-event', Lang.bind(this, this._onKeyPressEvent));
         this._group.connect('key-release-event', Lang.bind(this, this._onKeyReleaseEvent));
 
-        this.backgroundStack = new Shell.Stack();
+        this.backgroundStack = new St.Widget({ layout_manager: new Clutter.BinLayout() });
         this._backgroundBin = new St.Bin({ child: this.backgroundStack,
                                            x_fill: true, y_fill: true });
         this._monitorConstraint = new Layout.MonitorConstraint();
-- 
1.8.2.1

